tests/test_parseRecipe.py

```diff
@@ -57,8 +57,8 @@
 class Recoder(object):
   def __init__(self):
     self.buffer = ""
-  def __call__(self, s):
-    self.buffer += s
+  def __call__(self, s, *a):
+    self.buffer += s % a
```

Comment:
can you do `format(s, **kwds) % *a` instead, so that we can also use the keywords?

Reply:
I'm not a big fan of overriding the builtin `format` (which does something else), though `s % (kwds or a)` would work, I suppose...

The problem is that the built-in logger doesn't have that functionality, I think. It uses keywords for special situations (e.g. to tell it that it should print exception info attached to the log line). There might be a way to retrofit that, but it's likely a bit hacky, if not impossible.

Reply:
For the record, the reason why we have an override for the format method is that at the time we had to support python 2.5. If you want, feel free to replace all of them with `str.format`. That said, please, do not go and replace the current "keyword based" solution everywhere with positional formatting.

Given you replaced the keyword functionality only in an handful of places let's merge this.